### PR TITLE
Update facet configuration `position` and `optionsLimit` parameters

### PR DESCRIPTION
**Pull request by mocca102**
# Issue
- `position` and `options_limit` default to zero due to type `int`

# Fix
- Change type from `int` to `Integer` which will default the values to null instead of zero
- Add tests

### CODE REVIEW

File: constructorio-client/src/test/java/io/constructor/client/ConstructorIOFacetOptionConfigurationTest.java [hunk 1]
```diff
@@ -177,12 +177,12 @@ public void testCreateFacetOptionConfigurations() throws Exception {
         assertEquals("test-option", jsonOption1.get("value"));
         assertEquals("test-alias", jsonOption1.get("value_alias"));
         assertEquals("Test Option", jsonOption1.get("display_name"));
-        assertEquals(1, jsonOption1.get("position"));
+        assertEquals(Integer.valueOf(1), jsonOption1.get("position"));
```
Comment (stanlp1) [inline constructorio-client/src/test/java/io/constructor/client/ConstructorIOFacetOptionConfigurationTest.java:180]: Can we add a similar test to this where we set values to null, see if it gets set correctly?

**Review by stanlp1 — APPROVED** [review]
LGTM
